src/mlpack/methods/preprocess/preprocess_one_hot_encoding_main.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file preprocess_one_hot_encoding_main.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to do One-Hot Encoding on features from a dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/one_hot_encoding.hpp>
+
+PROGRAM_INFO("One Hot Encoding",
+    // Short description.
+    "A utility to do one hot encoding on features of dataset.",
+    // Long description.
+    "This utility takes a dataset and a vector of indices and does one hot "
+    "encoding of the respective features at those indices. "
+    "\n\n"
+    "The output matrices with encoded features may be saved with the " +
+    PRINT_PARAM_STRING("output") + " parameters."
+    "\n\n"
+    "So, a simple example where we want to encode 1st and 3rd feature"
+    " from dataset " + PRINT_DATASET("X") + " into " +
+    PRINT_DATASET("X_output") + " would be"
+    "\n\n" +
+    PRINT_CALL("preprocess_one_hot_encoding", "input", "X", "output",
+        "X_ouput", "indices", 1 , "indices", 3),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// Define parameters for data.
+PARAM_MATRIX_IN_REQ("input", "Matrix containing data.", "i");
+PARAM_MATRIX_OUT("output", "Matrix to save one hot encoded features "
+    "data to.", "o");
+
+PARAM_VECTOR_IN_REQ(size_t, "indices", "Index of Column which"
+    "need to be one hot encoded", "c");
```

Comment:
Actually (I deleted my earlier comment) I think that we have to use `int` here, not `size_t`.  So try this:

```suggestion
PARAM_VECTOR_IN_REQ(int, "indices", "Index of Column which"
    "need to be one hot encoded", "c");
```

I think that this will work---then you can check to see that all of the given values are non-negative in the `mlpackMain()` function. :+1: